Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -590,6 +590,15 @@ theorem map_prod_of_subset_primeFactors [CommSemiring R] {f : ArithmeticFunction
     f (∏ a ∈ t, a) = ∏ a ∈ t, f a :=
   map_prod_of_prime h_mult t fun _ a => prime_of_mem_primeFactors (ht a)
 
+theorem map_div_of_squarefree [CommGroupWithZero R] {f : ArithmeticFunction R}
+    (h_mult : IsMultiplicative f) {l d : ℕ} (hdl : d ∣ l) (hl : Squarefree l) (hd : f d ≠ 0) :
```

Comment:
Might it be better to relax `Squarefree l` to `(l / d).Coprime d` here? That allows things like `f (36 / 9)`